server/server.ts

```diff
@@ -14,6 +16,83 @@ const PORT = process.env.PORT || 8000;
 const app = express();
 app.use(express.json());
 app.use(express.static('../client'));
+app.use(require('morgan')('combined'));
+
+// From Passport example
+app.use(require('cookie-parser')());
+app.use(require('body-parser').urlencoded({ extended: true }));
+app.use(require('express-session')({
+  secret: 'keyboard cat', // TODO ?
+  resave: true,
+  saveUninitialized: true,
+}));
+
+app.use(passport.initialize());
+app.use(passport.session());
+
+
+passport.use(new Strategy(
+  {
+  clientID: process.env['FACEBOOK_CLIENT_ID'],
+  clientSecret: process.env['FACEBOOK_CLIENT_SECRET'],
+  callbackURL: '/return'
+  },
+  async function(accessToken, refreshToken, profile, done) {
+    const pgClient = await pgPool.connect();
+    try {
+      await pgClient.query('BEGIN');
+
+      let user = (await pgClient.query(`
+        SELECT *
+        FROM user_table
+        WHERE auth_provider = 'facebook'
+          AND id_from_provider = $1
+      `, [profile.id])).rows[0];
+
+      if (!user) {
+        user = (await pgClient.query(`
+          INSERT INTO
+            user_table(auth_provider, id_from_provider)
+          VALUES
+            ('facebook', $1)
+          RETURNING *
+        `, [profile.id])).rows[0]; // Should I store other profile fields?
```

Comment:
🛑 Let's store profile.displayName (as user_table.name) too

Reply:
Nah, that can wait -- not like I'm gonna have other users for now (only reason I want this right now is for db readability)